duneana/CAFMaker/CAFMaker_module.cc

```diff
@@ -175,299 +205,480 @@ namespace caf {
-
-        sr.CVNResult0Pions = (*cvnin)[0].Get0pionsProbability();
-        sr.CVNResult1Pions = (*cvnin)[0].Get1pionsProbability();
-        sr.CVNResult2Pions = (*cvnin)[0].Get2pionsProbability();
-        sr.CVNResultNPions = (*cvnin)[0].GetNpionsProbability();
-
-        sr.CVNResult0Pizeros = (*cvnin)[0].Get0pizerosProbability();
-        sr.CVNResult1Pizeros = (*cvnin)[0].Get1pizerosProbability();
-        sr.CVNResult2Pizeros = (*cvnin)[0].Get2pizerosProbability();
-        sr.CVNResultNPizeros = (*cvnin)[0].GetNpizerosProbability();
-
-        sr.CVNResult0Neutrons = (*cvnin)[0].Get0neutronsProbability();
-        sr.CVNResult1Neutrons = (*cvnin)[0].Get1neutronsProbability();
-        sr.CVNResult2Neutrons = (*cvnin)[0].Get2neutronsProbability();
-        sr.CVNResultNNeutrons = (*cvnin)[0].GetNneutronsProbability();
+  //------------------------------------------------------------------------------
+
+  void CAFMaker::FillEnergyInfo(caf::SRNeutrinoEnergyBranch &ErecBranch, const art::Event &evt) const
+  {
+    //Filling the reg CNN results
+    art::InputTag itag(fRegCNNLabel, "regcnnresult");
+    art::Handle<std::vector<cnn::RegCNNResult>> regcnn = evt.getHandle<std::vector<cnn::RegCNNResult>>(itag);
+    if(!regcnn.failedToGet() && !regcnn->empty()){
+        const std::vector<float>& cnnResults = (*regcnn)[0].fOutput;
+        ErecBranch.regcnn = cnnResults[0];
+    }
+    else{
+      std::cout << "Warning: " << itag << " does not correspond to a valid RegCNNResult product" << std::endl;
+    }
+
+    art::Handle<dune::EnergyRecoOutput> ereconue = evt.getHandle<dune::EnergyRecoOutput>(fEnergyRecoNueLabel);
+    art::Handle<dune::EnergyRecoOutput> ereconumu = evt.getHandle<dune::EnergyRecoOutput>(fEnergyRecoNumuLabel);
+
+
+    if(ereconue.failedToGet()){
+      std::cout << "Warning: " << fEnergyRecoNueLabel << " does not correspond to a valid EnergyRecoOutput product" << std::endl;
+    }
+    else{
+      ErecBranch.calo = ereconue->fNuLorentzVector.E();
+    }
+
+    if(ereconumu.failedToGet()){
+      std::cout << "Warning: " << fEnergyRecoNumuLabel << " does not correspond to a valid EnergyRecoOutput product" << std::endl;
+    }
+    else{
+      ErecBranch.lep_calo = ereconumu->fNuLorentzVector.E();
+    }
+
+        //TODO: See if we could add more fine-grained information here.
+
+
+    
+  }
+
+  //------------------------------------------------------------------------------
+
+  void CAFMaker::FillRecoParticlesInfo(caf::SRRecoParticlesBranch &recoParticlesBranch, const art::Event &evt) const
+  {
+    lar_pandora::PFParticleVector particleVector;
+    lar_pandora::LArPandoraHelper::CollectPFParticles(evt, fPandoraNuVertexModuleLabel, particleVector);
+    unsigned int nuID = 999;
+    for (unsigned int n = 0; n < particleVector.size(); ++n) {
+      const art::Ptr<recob::PFParticle> particle = particleVector.at(n);
+      if(particle->IsPrimary()){
+        nuID = n;
```

Comment:
ah, so the nuID is the index of the neutrino PFParticle in the PFParticle list. I _think_ it's better to use the `Self()` function to get a unique ID for the PFParticle. I know they're meant to be the same.. but just incase that particleVector is reordered at some point. You can never be too careful right?